src/components/StaffComments/StaffComments.vue

```diff
@@ -0,0 +1,256 @@
+        v-on="on"
+      >
+        <v-icon medium>mdi-comment-text-outline</v-icon>
+        <span>{{numComments}}</span>
+      </v-btn>
+    </template>
+
+    <!-- the menu (dialog) -->
+    <v-card flat id="staff-comment-container" class="px-8 py-6">
+      <v-card-title class="d-flex justify-space-between pa-0">
+        <div>
+          <v-icon medium color="primary">mdi-comment-text-outline</v-icon>
+          <span>{{numComments}}</span>
+        </div>
+        <v-btn icon large class="mr-n3" @click="close()">
+          <v-icon color="primary">mdi-close</v-icon>
+        </v-btn>
+      </v-card-title>
+
+      <v-card-text class="mt-2 pa-0">
+        <v-textarea
+          ref="textarea"
+          autofocus
+          no-resize
+          filled
+          rows="5"
+          v-model="comment"
+          placeholder="Enter Comments"
+          :rules="rules"
+        />
+      </v-card-text>
+
+      <v-card-actions class="d-flex justify-space-between pa-0">
+        <div class="body-2 mt-1">{{charsRemaining}}</div>
+        <div class="mr-n3">
+          <v-btn text color="primary" class="font-weight-bold" :loading="isSaving" @click="save()">
+            Save
+          </v-btn>
+          <v-btn text color="primary" @click="close()">
+            Cancel
+          </v-btn>
+        </div>
+      </v-card-actions>
+
+      <v-card-text class="mt-6 pa-0">
+        <div class="existing-comments pr-5">
+          <div v-for="(comment, i) in comments" :key="i" class="body-2">
+            <p class="pre-line" v-html="comment.comment" />
+            <p class="font-italic">
+              {{ comment.submitterDisplayName }}
+              &hyphen;
+              {{apiToSimpleDateTime(comment.timestamp)}}
+            </p>
+          </div>
+        </div>
+      </v-card-text>
+    </v-card>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop } from 'vue-property-decorator'
+import { CommentIF, FormIF } from '@/interfaces'
+import { DateMixin, FilingMixin } from '@/mixins'
+
+@Component({})
+export default class StaffComments extends Mixins(DateMixin, FilingMixin) {
+  $refs!: Vue['$refs'] & {
+    textarea: FormIF
+  }
+
+  /** Axios instance prop. */
+  @Prop({ required: true })
+  readonly axios: any
+
+  /** Business ID prop. */
+  @Prop({ required: true })
+  readonly businessId: string
+
+  /** Max Length prop (optional). */
+  @Prop({ default: 4096 })
+  readonly maxLength: number
+
+  /** Model property for v-menu (ie, whether to show the dialog). */
+  private showComments = false
+
+  /** The list of comments. */
+  private comments: Array<CommentIF> = []
+
+  /** The new comment's text. */
+  private comment: string = null
+
+  /** Whether a comment is being saved. */
+  private isSaving = false
+
+  /** The number of chars remaining in the new comment. */
+  private get charsRemaining (): number {
+    const length = this.comment ? this.comment.length : 0 // comment may be null
```

Comment:
I had to remove the previous conditional chaining (`a?.b`) as that caused a JS error. This is likely due to the TS version used but I can find nothing out of date.

(This is a future problem to fix.)